### ISSUE TITLE
[Feature Request: Linux] - A way to use "-noprofile" when running script directly from shell ( $> ./myscript.ps1 ) using #!/usr/bin/env pwsh 

### ISSUE THREAD

**Issue by MostHated** [PowerShell/PowerShell#15103]
Hey there,
    I apologize if there is a way to do this that I am not aware of, but I have not come across info on it if there is. 

My request is to have a way to run scripts directly from a shell without loading the items in my $PROFILE.

Currently, I have a PowerShell script I created running on Linux which I use to parse /proc/net/dev to determine the current network rate. As seen in the excerpt below, the script uses the shebang `#!/usr/bin/env pwsh` in order to allow me to invoke the script on a schedule easily from [Jobber](https://github.com/dshearer/jobber) as seen in the second dropdown below. There doesn't seem to be a way that I am aware of to run the script without the unnecessary overhead of including the loading of my $PROFILE, such as custom prompt and other modules that are only needed/used in an active terminal. 

I was attempting to profile the script by watching it in HTOP and noticed that upon each invocation it was somewhere in the neighborhood of 100MB in size and 27 threads, which I thought was a bit excessive. I went into my $PROFILE and commented everything out and right away saw each invocation dropped down to 87MB (still 27 threads). While still pretty heavy, definitely better. 

I realize that there is always going to be fairly substantial overhead of running a PowerShell script on Linux compared to simply running a Bash script due to the runtime and all that, but I feel it would definitely help with adoption to be able to run things in a familiar manner without having to exclusively choose between any sort of personal customization and trying to maintain minimal overhead for additional performance in these two situations (me, actively using pwsh in a terminal vs having a script run on a schedule). 

All in all, I love being able to have the much more friendly and familiar(to me, at least) coding experience of the .Net ecosystem on my Linux boxes and was hoping to utilize it more often and in different situations, but if each script is going to cost 100MB a pop, it seems that I my thought of replacing a lot of these small random bash scripts on my system with PowerShell scripts may not be the best idea and will end up being much more situational based on need. Either way, though, I really do like at least having the option and ability to do so, at least. :+1: 

Thanks,
-MH

<details>
<summary>Script Excerpt</summary>

```powershell
#!/usr/bin/env pwsh
<#	
	.NOTES
	======================================================
     Created on:   03/24/2021
     Created by:   instance.id
     Platform:     Linux
	 Filename:     traffic.ps1
	======================================================
	.DESCRIPTION
		Parses cat /proc/net/dev to determine current network transfer rate
#>

# --------------- Linux current network transfer rate check
# ---------------------------------------------------------
param (
    [Parameter()]
    [int]$Sleep = 1,
    [string]$Device,
    [switch]$Human,
    [switch]$NoLabel,
    [switch]$Pad,
    [switch]$Conky
)

```

</details>

<details>
<summary>Jobber Invocation</summary>

```yaml
jobs:
  speed:
     cmd: /home/mosthated/.conky/scripts/traffic.ps1 -Device enp6s0 -Sleep 1 -Human -Pad -Conky
     time: '*/2 * * * * *'
     onError: Continue
```

</details>

**Comment by mklement0** [comment]
It is unfortunate that PowerShell loads its profiles even in unattended (non-interactive) invocations, which not only makes for increased overhead but also for a less predictable execution environment.

The problem has been discussed before, and the idea came up to provide  _separate executable_ that doesn't load  profiles and fixes other legacy problems: #8072.

Unfortunately, momentum behind this has stalled, but I'm definitely in favor of resolving this.

**Comment by mklement0** [comment]
It is definitely a suboptimal solution, but if your scripts run on _Linux only_ and you're willing to assume a _default installation_, you can use `#!/usr/bin/pwsh -noprofile`

**Comment by MostHated** [comment]
Thanks for the reply. This script in particular is linux only as it is using cat /proc/net/dev to get the network traffic, but I did try `#!/usr/bin/env pwsh -noprofile` and that didn't work, but I didn't try `#!/usr/bin/pwsh -noprofile`, I will give that a go. Thanks.

**Comment by mklement0** [comment]
`#!/usr/bin/env pwsh -noprofile` doesn't work, because it uses _two_ arguments, which Linux doesn't support (macOS does, but this discrepancy prevents a _portable_ (cross-Unix) solution).

**Comment by jborean93** [comment]
As a workaround you can create an executable script in the `PATH` Called `pwsh_no_profile` which just calls `pwsh -NoProfile` and set that as your shebang argument.

**Comment by 237dmitry** [comment]
`#!/usr/bin/env -S pwsh -noprofile` and `chmod +x script` to run out from pwsh shell.

**Comment by mklement0** [comment]
Nice, @237dmitry - didn't know about the `-S` option for making `env` split the single argument it is given on Linux into individual arguments, but there's a caveat:

* On Linux distros, it requires [GNU coreutils v8.30+](https://savannah.gnu.org/forum/forum.php?forum_id=9187), which are included with Ubuntu 19.04+, for instance, but not with Ubuntu 18.04 and earlier.
In other words: older Linux distros may not support it (though that problem will take care of itself over time).

* (On macOS it is supported since at least Catalina; it isn't necessary there, but does no harm, which makes the solution portable.)

Just to spell out @jborean93's suggestion:

* Create the helper script, `pwsh-np` (from PowerShell) in `/usr/bin` (Linux) / `/usr/local/bin` (macOS):

```powershell
@'
#!/bin/sh
pwsh -noprofile "$@"
'@ | sudo pwsh -noprofile -c { $newBin = Join-Path ($IsLinux ? '/usr/bin' : '/usr/local/bin') pwsh-np; $input > $newBin; chmod a+x $newBin }
```

* Then you can use it in your shebang lines; e.g.:

```sh
#!/usr/bin/env pwsh-np

'Look, ma - no profile!'
```
